webapp/src/cljs/lipas/ui/map/views.cljs

```diff
@@ -243,29 +241,78 @@
     [mui/icon {:color "inherit" :font-size "medium"}
      "my_location"]]])
 
-(defn type-selector-single [{:keys [tr value on-change types]}]
-  (r/with-let [selected-type (r/atom value)]
-    (let [locale (tr)]
+(defn helper-select-onclick [type-code geom-help-open?]
+  (reset! geom-help-open? false) 
+  (==> [::sports-site-events/select-new-site-type type-code]))
+
+
+(defn helper-row [type-code type geom description geom-help-open?]
+  [mui/table-row {:on-click #(helper-select-onclick type-code geom-help-open?)}
+   [mui/table-cell type]
+   [mui/table-cell geom]
+   [mui/table-cell description]])
+
+(defn filter-by-term [term table-data]
+  (let [lower-case-term (lower-case term)]
+    (filter #(or (includes? (lower-case (% :name)) lower-case-term)
+                 (includes? (lower-case (% :geometry-type)) lower-case-term)
+                 (includes? (lower-case (% :description)) lower-case-term)
+                 (includes? (lower-case (join (% :tags))) lower-case-term))
+            table-data)))
+
+(defn type-helper-table [tr geom-help-open?]
+  (r/with-let [search-term (r/atom "")
+               table-data (<== [:lipas.ui.sports-sites.subs/type-table])]
+    (let [filtered-table-data (filter-by-term @search-term table-data)
+          sorted-and-filtered-table-data (sort-by :name filtered-table-data)]
+      [mui/grid {:container true}
```

Comment:
Grid container should contain grid items. https://v4.mui.com/components/grid/#grid

Please wrap the text-field and the table into `[mui/grid {:item true :xs 12} ...]`